testing/ef-tests/src/macros/sanity_slots.rs

```diff
@@ -1,63 +1,74 @@
+            use serde_yaml;
+            use tokio::sync::Mutex;
 
-                    for entry in std::fs::read_dir(&base_path).unwrap() {
-                        let entry = entry.unwrap();
-                        let case_dir = entry.path();
-                        if !case_dir.is_dir() {
-                            continue;
-                        }
+            use super::*;
 
-                        let case_name = case_dir.file_name().unwrap().to_str().unwrap();
-                        println!("Testing case: {}", case_name);
+            #[tokio::test]
+            async fn test_sanity_slots() {
+                let base_path = std::env::current_dir()
+                    .unwrap()
+                    .join("mainnet/tests/mainnet/deneb/sanity/slots/pyspec_tests");
 
-                        let slot: u64 = {
-                            let slot_path = case_dir.join("slots.yaml");
-                            let content = fs::read_to_string(slot_path)
-                                .expect("Failed to read slots.yaml");
+                for entry in std::fs::read_dir(&base_path).unwrap() {
+                    let entry = entry.unwrap();
+                    let case_dir = entry.path();
+                    if !case_dir.is_dir() {
+                        continue;
+                    }
 
-                            serde_yaml::from_str::<u64>(&content)
-                                .expect("Failed to parse slot number from slots.yaml")
-                        };
+                    let case_name = case_dir.file_name().unwrap().to_str().unwrap();
+                    println!("Testing case: {}", case_name);
 
-                        println!("Target slot: {}", slot);
+                    let slot: u64 = {
+                        let slot_path = case_dir.join("slots.yaml");
+                        let content =
+                            fs::read_to_string(slot_path).expect("Failed to read slots.yaml");
 
-                        let pre_state: BeaconState = utils::read_ssz_snappy(&case_dir.join("pre.ssz_snappy")).expect("cannot find test asset (pre.ssz_snappy)");
+                        serde_yaml::from_str::<u64>(&content)
+                            .expect("Failed to parse slot number from slots.yaml")
+                    };
 
-                        let mut state = pre_state.clone();
+                    let mut state: BeaconState =
+                        utils::read_ssz_snappy(&case_dir.join("pre.ssz_snappy"))
+                            .expect("cannot find test asset (pre.ssz_snappy)");
 
-                        let expected_post = utils::read_ssz_snappy::<BeaconState>(&case_dir.join("post.ssz_snappy"));
+                    let expected_post =
+                        utils::read_ssz_snappy::<BeaconState>(&case_dir.join("post.ssz_snappy"));
 
-                        let result = state.$processing_fn(state.slot + slot);
+                    let result = state.$process_slots_fn(state.slot + slot);
```

Comment:
```suggestion
                    let result = state.$process_slots_fn(state.slot + slot);
```

revert this back to what you have last time

Reply:
I got this error 
no rules expected `state_transition`
while trying to match end of macro

when i removed $processing_fn:path

Reply:
remove the argument when you call the macro, as you are removing the argument for the macro

Reply:
ohh ok my bad